src/pages/searchPage/searchPageView.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect, useCallback } from 'react';
+import { fetchSearchPage, getAllPlanets } from '../../shared/API';
+import { useParams, useNavigate } from 'react-router-dom'; // Import useNavigate
+import SearchBlock from '../../features/SearchBlock/SearchBlockView';
+import ResultsBlock from '../../features/ResultsBlock/ResultsBlockView';
+import ErrorButton from '../../features/ErrorButton/ErrorButtonView';
+
+const SearchPage = () => {
+  const [searchTerm, setSearchTerm] = useState(
+    localStorage.getItem('searchRequest') || ''
+  );
```

Comment:
` localStorage.getItem('searchRequest')` will be called on every render, it will not cause state change even  localStorage.getItem('searchRequest') returns new object on every call.

best practice to use function that will be called by use state once

https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

Reply:
how about this ?

`const getLastRequest = (): string => {
  return localStorage.getItem('searchRequest') || '';
};

const SearchPage = () => {
  const [searchTerm, setSearchTerm] = useState(getLastRequest);`

Reply:
great!